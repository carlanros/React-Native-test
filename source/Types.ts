
import * as login from './components/Login/types';

/**
 * All possible objects in Redux store
 *
 * @export
 * @interface IApplicationState
 */
export interface IApplicationState {
    login?: login.ILoginState;
}

/**
 * Describes possible props for a components.
 *
 * All actions and state objects might not be available.
 *
 * @export
 * @interface IProps
 */
export interface IProps extends IApplicationState {
    dispatch?: any;
    FacebookLogin?: login.FacebookLogin;
}