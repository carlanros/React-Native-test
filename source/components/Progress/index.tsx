import * as React from 'react';
import {View, Text, ActivityIndicator} from 'react-native';
interface IProps {
    amount: number;
}

const Progress = ({amount}: IProps) => {
    return (<View>
        <Text>Progress: {amount}</Text>
        <ActivityIndicator />
    </View>);
};
export default Progress;