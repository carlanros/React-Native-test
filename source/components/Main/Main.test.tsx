/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';

import { Main }from './'; // här
import * as types from '../../Types';
import  Progress  from '../Progress'; // upp 1 nivå

jest.unmock('react-native');

function setup() {
  const props: types.IProps = {
    login: {
      user: null,
      isLoggedIn: false,
    },

  };

  const enzymeWrapper = shallow(<Main {...props} /> );

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *   import { Text } from 'react-native';
   *    const busyProps = enzymeWrapper.find(Text).props();
   *    expect(busyProps.text).toBe('test);
   */
  describe('Main', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
      // Test Progress
      const progress = enzymeWrapper.find(Progress).props();
      expect(progress.amount).toBe(0);
    });
  });
});