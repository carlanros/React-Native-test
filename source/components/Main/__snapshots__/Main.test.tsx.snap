// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Main should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          style={
            Object {
              "fontSize": 40,
              "padding": 10,
            }
          }
        >
          Welcome 
        </UNDEFINED>,
        <UNDEFINED
          onChangeText={[Function]}
          placeholder="Type name"
          style={
            Object {
              "fontSize": 40,
              "height": 40,
            }
          }
        />,
        <Progress
          amount={0}
        />,
      ],
      "style": Object {
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Welcome ",
          "style": Object {
            "fontSize": 40,
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": "Welcome ",
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeText": [Function],
          "placeholder": "Type name",
          "style": Object {
            "fontSize": 40,
            "height": 40,
          },
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            style={
              Object {
                "fontSize": 40,
                "padding": 10,
              }
            }
          >
            Welcome 
          </UNDEFINED>,
          <UNDEFINED
            onChangeText={[Function]}
            placeholder="Type name"
            style={
              Object {
                "fontSize": 40,
                "height": 40,
              }
            }
          />,
          <Progress
            amount={0}
          />,
        ],
        "style": Object {
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Welcome ",
            "style": Object {
              "fontSize": 40,
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": "Welcome ",
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeText": [Function],
            "placeholder": "Type name",
            "style": Object {
              "fontSize": 40,
              "height": 40,
            },
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
